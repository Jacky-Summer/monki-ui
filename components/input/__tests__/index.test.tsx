import React from 'react'
import { mount } from 'enzyme'
import Input, { InputProps } from '../input'

const defaultProps: InputProps = {
  onChange: jest.fn(),
  placeholder: 'this is input',
}

describe('Input component', () => {
  it('should render the correct default Input', () => {
    const wrapper = mount(<Input {...defaultProps} />)
    expect(wrapper.find('input.mk-input-inner').prop('placeholder')).toEqual(
      'this is input',
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should support size', () => {
    const wrapper = mount(<Input {...defaultProps} size="lg" />)
    expect(wrapper.find('.mk-input-wrapper').hasClass('input-size-lg')).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the disabled Input on disabled property', () => {
    const wrapper = mount(<Input {...defaultProps} disabled />)
    expect(wrapper.find('.mk-input-wrapper').hasClass('is-disabled')).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render prefix and suffix element on prefix/suffix property', () => {
    const wrapper = mount(<Input {...defaultProps} prefix="https://" suffix=".com" />)
    expect(wrapper.find('.mk-input-group-prefix').text()).toEqual('https://')
    expect(wrapper.find('.mk-input-group-suffix').text()).toEqual('.com')
    expect(wrapper).toMatchSnapshot()
  })
})
