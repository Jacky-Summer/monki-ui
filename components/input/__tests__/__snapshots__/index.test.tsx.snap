// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render prefix and suffix element on prefix/suffix property 1`] = `
<ForwardRef
  onChange={[MockFunction]}
  placeholder="this is input"
  prefix="https://"
  suffix=".com"
>
  <div
    className="mk-input-wrapper input-group input-group-suffix input-group-prefix"
  >
    <div
      className="mk-input-group-prefix"
    >
      https://
    </div>
    <input
      className="mk-input-inner"
      onChange={[MockFunction]}
      placeholder="this is input"
    />
    <div
      className="mk-input-group-suffix"
    >
      .com
    </div>
  </div>
</ForwardRef>
`;

exports[`Input component should render the correct default Input 1`] = `
<ForwardRef
  onChange={[MockFunction]}
  placeholder="this is input"
>
  <div
    className="mk-input-wrapper"
  >
    <input
      className="mk-input-inner"
      onChange={[MockFunction]}
      placeholder="this is input"
    />
  </div>
</ForwardRef>
`;

exports[`Input component should render the disabled Input on disabled property 1`] = `
<ForwardRef
  disabled={true}
  onChange={[MockFunction]}
  placeholder="this is input"
>
  <div
    className="mk-input-wrapper is-disabled"
  >
    <input
      className="mk-input-inner"
      disabled={true}
      onChange={[MockFunction]}
      placeholder="this is input"
    />
  </div>
</ForwardRef>
`;

exports[`Input component should support size 1`] = `
<ForwardRef
  onChange={[MockFunction]}
  placeholder="this is input"
  size="lg"
>
  <div
    className="mk-input-wrapper input-size-lg"
  >
    <input
      className="mk-input-inner"
      onChange={[MockFunction]}
      placeholder="this is input"
    />
  </div>
</ForwardRef>
`;
